cloudinit/util.py

```diff
@@ -1667,10 +1667,12 @@ def mount_cb(device, callback, data=None, mtype=None,
             mtypes = ["auto"]
     elif platsys.endswith("bsd"):
         if mtypes is None:
-            mtypes = ['ufs', 'cd9660', 'vfat']
+            mtypes = ['ufs', 'cd9660', 'msdos']
         for index, mtype in enumerate(mtypes):
             if mtype == "iso9660":
                 mtypes[index] = "cd9660"
+            if mtype in ["vfat", "msdosfs", "msdos"]:
```

Comment:
We don't need `"msdos"` in this list; `mtypes[index]` is already `"msdos"` in that case.

```suggestion
            if mtype in ["vfat", "msdosfs"]:
```